src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -280,6 +281,15 @@ private void validateColumnType(Errors errors, ColumnModel columnModel, String k
     }
   }
 
+  private void validateColumnMode(Errors errors, ColumnModel columnModel) {
+    if (Boolean.TRUE.equals(columnModel.isRequired()) && columnModel.isArrayOf()) {
```

Comment:
The boolean check caused me to do a double-take, nice find!  Is there a reason why `columnModel.isArrayOf()` does not require the same kind of check?

This might be worth a comment summarizing why `if (columnModel.isRequired() && columnModel.isArrayOf())` does not work.

Reply:
The `isArrayOf` field is set to `false` by default in the open-api spec, but `isRequired` does not have a default value and could be `null` (it has a default `false` value in the database). So this explicitly checks if `isRequired` is true to avoid a null pointer exception.